pkg/resources/pods.go

```diff
@@ -75,6 +75,19 @@ func (pc PodAccessor) PodIPsByAge() ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Keep only running ones.
+	write := 0
+	for i := range pods {
+		p := pods[i]
+		if p.Status.Phase == corev1.PodRunning && p.DeletionTimestamp == nil {
+			if write != i {
+				pods[write] = p
+			}
+			write++
+		}
+	}
+	pods = pods[:write]
```

Comment:
I like this pattern for in-place filters! Will add it to my back-of-head toolbox 😂

Reply:
There's also a non order preserving one, a bit more effective, since it minimizes copies :)
```python
rw,  l = 0,  len(x)
while rw < len(x):
  if keep(x[rw]):
    rw+=1
  else:
    x[rw] = x[l-1]
    l-=1
return x[:rw]
```